src/coreclr/nativeaot/Bootstrap/main.cpp

```diff
@@ -181,13 +182,20 @@ __attribute__((weak)) void _initialize()
 {
 }
 
+// Guard the "_initialize" call so that well-behaving hosts do not get affected by this workaround.
+bool g_CalledInitialize = false;
+struct WasiInitializationFlag { WasiInitializationFlag() { *(volatile bool*)&g_CalledInitialize = true; } };
+WasiInitializationFlag g_WasiInitializationFlag;
 #endif // TARGET_WASI
 #endif // !NATIVEAOT_DLL
 
 static int InitializeRuntime()
 {
 #if defined(NATIVEAOT_DLL) && defined(TARGET_WASI)
-    _initialize();
+    if (!g_CalledInitialize)
```

Comment:
Perhaps it would be simpler to have `static bool alreadyInitialized = false;` inside this function instead of separate struct that relies on static constructor? (assuming this function is called from inside `_initialize`)

Reply:
Ah ok, sorry.

Reply:
No problem, thank you for the review!